jenkins/databricks/test.sh

```diff
@@ -82,6 +82,10 @@ case "$BASE_SPARK_VERSION" in
         # Available versions https://repo.maven.apache.org/maven2/org/apache/iceberg/iceberg-spark-runtime-3.3_2.12/
         sw_versions[ICEBERG]=${ICEBERG_VERSION:-'0.14.1'}
         ;;
+    "3.3.2")
```

Comment:
I have a question on this, we don't officially support Iceberg for Spark versions other than 3.2.x, yet we have been adding these dependencies **and** testing Iceberg in Databricks 11.3 and 12.2 (Spark 3.3.0 and 3.3.2 respectively). Does this even make sense anymore given we don't even run these tests against vanilla Spark 3.3 and above?